somexchanger99/erp_utils.py

```diff
@@ -14,6 +14,58 @@ class ErpUtils(object):
     def __init__(self):
         self._client = Client(**settings.ERP_CONF)
 
+    def generate_e101_attachments(self, model, date, process, step, **kwargs):
+
+        date_to = kwargs.get('date_to')
+
+        polissa_obj = self._client.model('giscedata.polissa')
+        partner_obj = self._client.model('res.partner')
+        sw_obj = self._client.model('giscedata.switching')
+        sw_step_obj = self._client.model('giscedata.switching.step')
+        swe101 = self._client.model('giscedata.switching.e1.01')
+        switching_wizard = self._client.model('giscedata.switching.wizard')
+
+        e1s_objects = self._get_objects_with_attachment(
+            model, date, process=process, step=step, date_to=date_to
+        )
+        e101_attachments = []
+
+        for e1 in e1s_objects:
+            if not e1.polissa_ref_id:
+                continue
+
+            steps = sw_step_obj.search([('proces_id','=', 11)])
+            pas_ids = swe101.search([('sw_id', '=', e1.id)])
+            if not pas_ids:
+                continue
+
+            pas_id = pas_ids[0]
+            e101_step = sorted(steps)[0]
+            step_id = str( (e101_step, pas_id) )
+
+            polissa_id = e1.polissa_ref_id.id
+            titular_id = polissa_obj.read(polissa_id, ['titular'])
+            lang = partner_obj.read(titular_id['titular'][0], ['lang'])['lang']
+
+            id_wiz = switching_wizard.create(
+                {'step_id': step_id, 'mark_as_processed': 0},
+                {'lang': lang, 'active_ids': [ e1.id], 'active_id':  e1.id}
+            )
+
+            name = 'e101.xml'
+
+            switching_wizard.action_exportar_xml([id_wiz.id], {'active_ids': [ e1.id], 'active_id':  e1.id})
+
+            attachments = switching_wizard.read(
+                [id_wiz.id],
+                ['file'],
+                {'lang': lang, 'bin_size': False, 'tz': 'Europe/Madrid', 'active_ids': [e1.id], 'active_id': e1.id}
+            )
+        
+            e101_attachments.extend([{'name': name, 'datas': attachment['file']} for attachment in attachments])
```

Comment:
:bangbang:
like name is the same for all attachments, its possible that the process that upload this attachatments to the sftp, overwrite them! :hushed:

Reply:
Oops, we forgot to build it. Fixed.